dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java

```diff
@@ -1034,6 +1035,51 @@ public void getAllRelationships () throws DotSecurityException, DotDataException
         }
     }
 
+    @Test
+    public void testCreateSelfJoinedRelationship_contentletAddedAsChild () throws DotDataException, DotSecurityException {
+
+        //Get Default Language
+        Language language = languageAPI.getDefaultLanguage();
+
+        //Create a Structure
+        Structure structure = createStructure( "JUnit Test Structure_" + String.valueOf( new Date().getTime() ), "junit_test_structure_" + String.valueOf( new Date().getTime() ) );
+
+        //Create the Contentlets
+        Contentlet parent = createContentlet( structure, language, false );
+        Contentlet child = createContentlet( structure, language, false );
+
+        //Create the Self contained relationship
+        Relationship selfRelationship = createRelationShip(structure.getInode(), structure.getInode(), false );
+
+        //Create the contentlet relationships
+        List<Contentlet> contentRelationships = new ArrayList<>();
+        contentRelationships.add(child);
+
+        //Relate the content
+        contentletAPI.relateContent(parent, selfRelationship, contentRelationships, user, false );
+
+        //Find all the relationships for this contentlet
+        ContentletRelationships contentletRelationships = contentletAPI.getAllRelationships( parent.getInode() , user, false);
+
+        //Validations
+        assertNotNull(contentletRelationships );
+        assertTrue(contentletRelationships.getRelationshipsRecords() != null && !contentletRelationships.getRelationshipsRecords().isEmpty() );
+        for (ContentletRelationshipRecords record : contentletRelationships.getRelationshipsRecords()) {
+            if (record.isHasParent()) {
+                assertNotNull(record.getRecords());
+            }
+        }
+
+        //Clean up
+        contentletAPI.archive(child, user, false);
```

Comment:
This clean up should be in a finally section

Reply:
yeah good point.